includes/utils/class-amp-dom-utils.php

```diff
@@ -108,6 +108,18 @@ function( $matches ) {
 			);
 		}
 
+		/*
+		 * Add a pre-HTML5-style declaration of the encoding since libxml<2.8 doesn't recognize
+		 * HTML5's meta charset. See <https://bugzilla.gnome.org/show_bug.cgi?id=655218>.
+		 */
+		if ( version_compare( LIBXML_DOTTED_VERSION, '2.8', '<' ) ) {
+			$document = preg_replace(
+				'#<meta[^>]+charset="([^"]+)"#i',
+				'<meta http-equiv="Content-Type" content="text/html; charset=$1" id="meta-http-equiv">$0',
```

Comment:
Why the `$0`?

Reply:
Because I'm not replacing the original `meta charset` tag, just inserting the new `meta` tag before it. (That's also why I don't include the `>` in he regex, per your previous comment.)

I see that at a glance it looks like the aim is to replace the whole original meta tag, so I should probably add a comment to clarify what's going on.

Reply:
Thanks. I understand now. In that case you might want to use a lookahead match instead, like so:

```php
$document = preg_replace(
    '#(?=<meta[^>]+charset="([^"]+)")#i',
    '<meta http-equiv="Content-Type" content="text/html; charset=$1" id="meta-http-equiv">',
    $document
);
```

Not really a big deal.

Reply:
Oh, yeah, I should have thought of that. My only defense is that it was after midnight. :-P